airflow/providers/google/cloud/hooks/gcs.py

```diff
@@ -344,7 +344,7 @@ def provide_file(
         self,
         bucket_name: str = PROVIDE_BUCKET,
         object_name: Optional[str] = None,
-        object_url: Optional[str] = None,
+        dir: Optional[str] = None,
```

Comment:
Can you tell a little more about it? `object_url` parameter is used by `_fallback_object_url_to_object_name_and_bucket_name` decorator, so if we want the correct typing then we should add this parameter back.

Reply:
You are right. I missed that the decorator was changing the signature and that this was used for typing.

I reverted this change.